### ISSUE TITLE
Increase code coverage to an acceptable level

### ISSUE THREAD

**Issue by fmorency** [liftedinit/manifest-ledger#26]
Code coverage currently sits at 35.7%, which is low.

Edit: `poa` and `tokenfactory` modules coverage sit at **53.9%** and **39.9%** respectively. See https://github.com/strangelove-ventures/poa/pull/148 and https://github.com/strangelove-ventures/tokenfactory/pull/5

**Comment by fmorency** [comment]
The `interchaintest` tests are calling the `manifestd` a lot.

Idea: Instrument `manifestd` with coverage support and retrieve the coverage data. Merge this coverage data with unit test coverage data.

It works when using bare Docker but not via `interchaintest`

I'm not sure how to continue. 

- I modified `interchaintest` to propagate chain environment variables so that `GOCOVERDIR` behaves correctly: https://github.com/fmorency/interchaintest/tree/chain-env
- I added a method to retrieve the coverage data at the end of a given test (using `client.CopyFromContainer()`)

The problem is that the retrieved coverage data is empty.

I *think* I need a mechanism to be able to retrieve the coverage data from every temporary container spawned by `interchaintest` and not only from the "main" test container.
